src/models/data.py

```diff
@@ -200,27 +214,31 @@ def ds_folder_to_np(self,
 
             if clear_nans:
                 latlons = latlons[notnan_indices]
-            return ModelArrays(x=x_np, y=y_np, x_vars=list(x.data_vars),
+
+            x_data = TrainData(historical=x_np, pred_months=x_months)
+            return ModelArrays(x=x_data, y=y_np, x_vars=list(x.data_vars),
                                y_var=list(y.data_vars)[0], latlons=latlons)
 
-        return ModelArrays(x=x_np, y=y_np, x_vars=list(x.data_vars),
+        x_data = TrainData(historical=x_np, pred_months=x_months)
+        return ModelArrays(x=x_data, y=y_np, x_vars=list(x.data_vars),
                            y_var=list(y.data_vars)[0])
 
 
 class _TrainIter(_BaseIter):
 
-    def __next__(self) -> Tuple[Union[np.ndarray, torch.Tensor],
+    def __next__(self) -> Tuple[Tuple[Union[np.ndarray, torch.Tensor],
+                                      Union[np.ndarray, torch.Tensor]],
                                 Union[np.ndarray, torch.Tensor]]:
 
         if self.idx < self.max_idx:
-            out_x, out_y = [], []
+            out_x, out_x_add, out_y = [], [], []
```

Comment:
this is where we have the power with the `additional` vectors? (i.e. for the `current` timesteps in the `nowcast`)

Reply:
yes